### PR TITLE
feat-added signup feature

### LINKED ISSUE #15: Currently user can only login via hardcoded crendentials
To fix this we will add a signup feature. 
User can enter their credentials like name, email, and password.

### PR DESCRIPTION
**Pull request by manish591**
this fixes #15

### CODE REVIEW

File: src/hooks/useAuth.js [hunk 1]
```diff
@@ -1,48 +1,94 @@
...
-      const res = await axios.post('/api/auth/login', {
-        email, 
-        password
+      const res = await axios.post("/api/auth/login", {
+        email,
+        password,
       });
-      if(res.status === 200 || res.status === 201) {
-        setCurrentUser(res.data);
+      if (res.status === 200 || res.status === 201) {
```
Comment (romabulani) [inline src/hooks/useAuth.js:26]: response returned by login is 200, I think there is no need ti check for 201

File: src/hooks/useKeepAuthData.js [hunk 2]
```diff
@@ -1,28 +1,40 @@
...
   useEffect(() => {
-    localStorage.setItem("token", JSON.stringify(currentUser));
+    localStorage.setItem("user", JSON.stringify(currentUser));
     localStorage.setItem("login", isUserLogedIn);
-  }, [isUserLogedIn, currentUser]);
+    localStorage.setItem("token", JSON.stringify(myToken));
+
+    // stateDispatch({ type: 'GET_CART_DATA', payload: currentUser.cart });
+    // stateDispatch({ type: 'GET_WISHLIST_DATA', payload: currentUser.wishlist });
+
```
Comment (romabulani) [inline src/hooks/useKeepAuthData.js:36]: Remove these comments if not required

**Review by romabulani — APPROVED** [review]
- Clean code
- I can see you have used context in custom hooks instead of passing params as suggested yesterday

One small sugestion, if you have time, in catch block of network calls, instead of only console logging error, you can write function name, it helps in debugging and readability. eg : `console.log('in Login Handler', e)`

**Review by shrey27 — APPROVED** [review]
Nice refactor
PR is good to merge

File: src/hooks/useKeepAuthData.js (hunk 2 above)
Comment (manish591) [reply src/hooks/useKeepAuthData.js:36]: Will do the changes in refactor branch

**Pull request merged**